Do not install bins on system install

zmughal opened the pull request.

yoe committed:
Don't break system utilities on use of bins property

When alien_version_check is set, we may end up depending on the
system-installed version of the built package. In that case, the
absolute path in the dist_dir won't be available, and the binaries fail
to work.

Check for that using the ConfigData package, and use the binary in
/usr/bin directlry, assuming it's installed there.

zmughal committed:
Do not install bins wrappers on system install

zmughal committed:
Add test of bin wrappers for system and share

zmughal committed:
Die if wrapper exists in non-share install

zmughal commented:
@yoe, @plicease, if you get a chance, could you review this?

plicease commented on lib/Dist/Zilla/Plugin/Alien.pm:
```diff
@@ -505,19 +506,30 @@ use strict;
 use Path::Class;
-
-my $abs = file(dist_dir('{{ $dist->name }}'),'bin','{{ $bin }}')->cleanup->absolute;
+use File::Which;
```
We should add File::Which as a prerequisite when using this option, if not already added.  (gets pulled in already by AB, but we shouldn’t rely on that here).

zmughal replied:
Ah, that needs to go away. I removed the code in the else-branch that used `which()`.

plicease reviewed (approved):
I think this is a big improvement, I’d personally like to see the generated installer only generated when we need it, though on the other hand that increases the complexity so it isn’t for free.

zmughal committed:
Only use custom Module::Build if bins property set

Document this case and throw error if not possible to combine custom
`mb_class` with `bins`.

zmughal commented on lib/Dist/Zilla/Plugin/Alien.pm:
```diff
@@ -488,7 +489,7 @@ sub register_prereqs {
 
 has "+mb_class" => (
-	default => 'Alien::Base::ModuleBuild',
+	default => 'MyModuleBuild',
```
Done.
